File: app/src/main/java/com/manddprojectconsultant/screencam/SettingActivity.java
package com.manddprojectconsultant.screencam;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;
public class SettingActivity extends AppCompatActivity implements CameraPreviewFragment.ItemClickListenercamerapreview,CameraFacingFragment.ItemClickListenercamerafacing ,OrientationFragment.ItemClickListenerorientation, FPSFragment.ItemClickListenerfps, QualitySheetFragment.ItemClickListenervideo, ResolutionSheetDialog.ItemClickListener {
    public static final int REQUEST_CODE_INTRO = 1;
    public static final String PREF_KEY_FIRST_START = "com.heinrichreimersoftware.materialintro.demo.PREF_KEY_FIRST_START";
    private RelativeLayout llResolution, llQuality, llFPS, llOrientation, llRecordAudio, llFileLocation, llCountDown,
            llCamera, llCameraPreview, llShowBubble,llRating,llShare;
    private TextView tvResolution, tvQuality, tvFPS, tvOrientation, tvCamera, tvCameraPreview;
    private Switch swRecordAudio, swCountDown, swAppIntro, swShowBubble;
    ImageView llAppIntro, ivbackbutton;
    //final String[] list = getActivity().getResources().getStringArray(R.array.Choice_items);
    String[] ResolutionList, QualityList, FPSList, OrientationList, CameraList, CameraListPreview;
    public static final String Resolution = "Resolution";
    public static final String VideoQuality = "Quality";
    public static final String FPS = "FPS";
    public static final String Orientation = "Orientation";
    public static final String CameraFacing= "CameraFacing";
    public static final String CameraFrame= "CameraFrame";
    public static final String CameraPreview = "CameraPreview";


   // public static final String RecordAudio = "record audio";




    QualitySheetFragment qualitySheetFragment;
    ResolutionSheetDialog resolutionSheetDialog;
    FPSFragment fpsFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);
        findAllView();


        //Resolution Preference
        String resolution = SPVariables.getString(Resolution, SettingActivity.this);
        tvResolution.setText(resolution);

        //Video Quality Preference
        String videoquality = SPVariables.getString(VideoQuality, SettingActivity.this);
        tvQuality.setText(videoquality);

        //FPS preference
        String fpspre = SPVariables.getString(FPS, SettingActivity.this);
        tvFPS.setText(fpspre);

        //Orientation preference
        String orientationpre = SPVariables.getString(Orientation, SettingActivity.this);
        tvOrientation.setText(orientationpre);

        //CameraFacing preference
        String camerafacingpre = SPVariables.getString(CameraFacing, SettingActivity.this);
        tvCamera.setText(camerafacingpre);


        //CameraPreview Preference
        String camerapreviewpre = SPVariables.getString(CameraPreview, SettingActivity.this);
        tvCameraPreview.setText(camerapreviewpre);





        setTitle("Settings");
        getAllList();
        loadSettingDataFromSP();
        allOnClicks();
    }

    private void loadSettingDataFromSP() {
        //Resolution
        String aa = SPVariables.getString(Resolution,SettingActivity.this);
        if (SPVariables.getString("Resolution", SettingActivity.this).isEmpty()) {
            tvResolution.setText(aa);
        } else {
            //tvResolution.setText(SPVariables.getString("resolution", SettingActivity.this));
            tvResolution.setText(SPVariables.getString("Resolution", SettingActivity.this));
        }

        //Video Quality

        String videoquality = SPVariables.getString(VideoQuality, SettingActivity.this);
        if (SPVariables.getString("Quality", SettingActivity.this).isEmpty()) {
            tvQuality.setText(videoquality);
        } else {
            tvQuality.setText(SPVariables.getString("Quality", SettingActivity.this));
        }

        //FPS
        String fpspre = SPVariables.getString(FPS, SettingActivity.this);
        if (SPVariables.getString("FPS", SettingActivity.this).isEmpty()) {
            tvFPS.setText(fpspre);
        } else {
            tvFPS.setText(SPVariables.getString("FPS", SettingActivity.this));
        }

        //Orientation
        String orientationpre = SPVariables.getString(Orientation, SettingActivity.this);
        if (SPVariables.getString("Orientation", SettingActivity.this).isEmpty()) {
            tvOrientation.setText(orientationpre);
        } else {
            tvOrientation.setText(SPVariables.getString("Orientation", SettingActivity.this));
        }

        //Camera Facing
        String camerafacingpre = SPVariables.getString(CameraFacing, SettingActivity.this);
        if (SPVariables.getString("CameraFacing", SettingActivity.this).isEmpty()) {
            tvCamera.setText(camerafacingpre);
        } else {
            tvCamera.setText(SPVariables.getString("CameraFacing", SettingActivity.this));
        }

        //Camera Preview
        String camerapreviewpre = SPVariables.getString(CameraPreview, SettingActivity.this);
        if (SPVariables.getString("CameraPreview", SettingActivity.this).isEmpty()) {
            tvCameraPreview.setText(camerapreviewpre);
        } else {
            tvCameraPreview.setText(SPVariables.getString("CameraPreview", SettingActivity.this));
        }

        if (SPVariables.getString("RecordAudio", SettingActivity.this).equals("TRUE")) {
            swRecordAudio.setChecked(true);
        } else {
            swRecordAudio.setChecked(false);
        }

        if (SPVariables.getString("CountDown", SettingActivity.this).equals("TRUE")) {
            swCountDown.setChecked(true);
        } else {
            swCountDown.setChecked(false);
        }
//        if (SPVariables.getString("AppIntro", SettingActivity.this).equals("TRUE")) {
//            swAppIntro.setChecked(true);
//        } else {
//            swAppIntro.setChecked(false);
//        }
        if (SPVariables.getString("ShowBubble", SettingActivity.this).equals("TRUE")) {
            swShowBubble.setChecked(true);
        } else {
            swShowBubble.setChecked(false);
        }
    }

    private void allOnClicks() {
        llResolution.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(ResolutionList, "Resolution", tvResolution.getText().toString());
                SingleChoiceDialog.setCancelable(true);
                SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
                resolutionSheetDialog = new ResolutionSheetDialog();
                resolutionSheetDialog.show(getSupportFragmentManager(), "Resolution");
            }
        });
        llQuality.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
           /*     DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(QualityList, "Quality", tvQuality.getText().toString());
                SingleChoiceDialog.setCancelable(true);
                SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
                qualitySheetFragment = new QualitySheetFragment();
                qualitySheetFragment.show(getSupportFragmentManager(), "Quality");
            }
        });
        llFPS.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(FPSList, "FPS", tvFPS.getText().toString());
                SingleChoiceDialog.setCancelable(true);
                SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
                fpsFragment = new FPSFragment();
                fpsFragment.show(getSupportFragmentManager(), "FPS");
            }
        });
        llOrientation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               /* DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(OrientationList, "Orientation", tvOrientation.getText().toString());
                SingleChoiceDialog.setCancelable(true);
                SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/

                OrientationFragment orientationFragment=new OrientationFragment();
                orientationFragment.show(getSupportFragmentManager(),"Orientation");


            }
        });
        llCamera.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
        /*        DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(CameraList, "Camera", tvCamera.getText().toString());
                SingleChoiceDialog.setCancelable(true);
                SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/

                CameraFacingFragment cameraFacingFragment=new CameraFacingFragment();
                cameraFacingFragment.show(getSupportFragmentManager(),"CameraFacing");


            }
        });
        llCameraPreview.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
        /*        DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(CameraListPreview, "CameraPreview", tvCameraPreview.getText().toString());
                SingleChoiceDialog.setCancelable(true);
                SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/

                CameraPreviewFragment cameraPreviewFragment=new CameraPreviewFragment();
                cameraPreviewFragment.show(getSupportFragmentManager(),"Camera Preview");

            }
        });






        swRecordAudio.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (isChecked) {
                    SPVariables.setString("RecordAudio", "TRUE", SettingActivity.this);
                } else {
                    SPVariables.setString("RecordAudio", "FALSE", SettingActivity.this);
                }
            }
        });
        swCountDown.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (isChecked) {
                    SPVariables.setString("CountDown", "TRUE", SettingActivity.this);
                } else {
                    SPVariables.setString("CountDown", "FALSE", SettingActivity.this);
                }
            }
        });
        llAppIntro.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                PrefManager prefManager = new PrefManager(SettingActivity.this);
                prefManager.setFirstTimeLaunch(true);
                Intent intent = new Intent(SettingActivity.this, Welcome_Screen.class);
                startActivityForResult(intent, REQUEST_CODE_INTRO);
                //finish();
            }
        });
//        swAppIntro.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                if (isChecked) {
//                    SPVariables.setString("AppIntro", "TRUE", SettingActivity.this);
//                } else {
//                    SPVariables.setString("AppIntro", "FALSE", SettingActivity.this);
//                }
//            }
//        });
        swShowBubble.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (isChecked) {
                    SPVariables.setString("ShowBubble", "TRUE", SettingActivity.this);
                } else {
                    SPVariables.setString("ShowBubble", "FALSE", SettingActivity.this);
                }
            }
        });
        ivbackbutton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent(SettingActivity.this, MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
        startActivity(intent);
        overridePendingTransition(0, 0);
        finish();
        super.onBackPressed();
    }

    private void getAllList() {
        ResolutionList = getResources().getStringArray(R.array.Resolution);
        QualityList = getResources().getStringArray(R.array.Quality);
        FPSList = getResources().getStringArray(R.array.FPS);
        OrientationList = getResources().getStringArray(R.array.Orientation);
        CameraList = getResources().getStringArray(R.array.Camera);
        CameraListPreview = getResources().getStringArray(R.array.CameraPreview);
    }

    private void findAllView() {
        ivbackbutton = findViewById(R.id.ivbackbutton);
        llResolution = findViewById(R.id.llResolution);
        llQuality = findViewById(R.id.llQuality);
        llFPS = findViewById(R.id.llFPS);
        llOrientation = findViewById(R.id.llOrientation);
        llRecordAudio = findViewById(R.id.llRecordAudio);
        llFileLocation = findViewById(R.id.llFileLocation);
        llCountDown = findViewById(R.id.llCountDown);
        llCamera = findViewById(R.id.llCamera);
        llCameraPreview = findViewById(R.id.llCameraPreview);
        llAppIntro = findViewById(R.id.llAppIntro);
        llShowBubble = findViewById(R.id.llShowBubble);
        tvResolution = findViewById(R.id.tvResolution);
        tvQuality = findViewById(R.id.tvQuality);
        tvFPS = findViewById(R.id.tvFPS);
        tvOrientation = findViewById(R.id.tvOrientation);
        tvCamera = findViewById(R.id.tvCamera);
        tvCameraPreview = findViewById(R.id.tvCameraPreview);
        swRecordAudio = findViewById(R.id.swRecordAudio);
        swCountDown = findViewById(R.id.swCountDown);
        //swAppIntro = findViewById(R.id.swAppIntro);
        swShowBubble = findViewById(R.id.swShowBubble);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_CODE_INTRO) {
            if (resultCode == RESULT_OK) {
                PreferenceManager.getDefaultSharedPreferences(this).edit()
                        .putBoolean(PREF_KEY_FIRST_START, false)
                        .apply();
            } else {
                PreferenceManager.getDefaultSharedPreferences(this).edit()
                        .putBoolean(PREF_KEY_FIRST_START, true)
                        .apply();
                //User cancelled the intro so we'll finish this activity too.
                //finish();
            }
        }
    }

    @Override
    public void onItemClick(String item) {
        String resolution = item.toString();
        SPVariables.setString(Resolution, resolution, SettingActivity.this);
        startActivity(new Intent(SettingActivity.this, SettingActivity.class).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION));
        finish();
        overridePendingTransition(0, 0);
        //tvResolution.setText(resolution);
    }

    @Override
    public void onItemClickvideo(String videoquality) {
        String vidquality = videoquality.toString();
        SPVariables.setString(VideoQuality, vidquality, SettingActivity.this);
        startActivity(new Intent(SettingActivity.this, SettingActivity.class).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION));
        finish();
        overridePendingTransition(0, 0);
    }

    @Override
    public void onItemClickfps(String fps) {
        String fp = fps.toString();
        SPVariables.setString(FPS, fp, SettingActivity.this);
        startActivity(new Intent(SettingActivity.this, SettingActivity.class).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION));
        finish();
        overridePendingTransition(0, 0);
    }

    @Override
    public void onItemClickorientation(String orientation) {
        String orien = orientation.toString();
        SPVariables.setString(Orientation, orien, SettingActivity.this);
        startActivity(new Intent(SettingActivity.this, SettingActivity.class).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION));
        finish();
        overridePendingTransition(0, 0);
    }

    @Override
    public void onItemClickcamerafacing(String camerafacing) {
        String camera = camerafacing.toString();
        SPVariables.setString(CameraFacing, camera, SettingActivity.this);
        startActivity(new Intent(SettingActivity.this, SettingActivity.class).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION));
        finish();
        overridePendingTransition(0, 0);


    }

    @Override
    public void onItemClickcamerapreview(String camerapreview) {

        String camerapre = camerapreview.toString();

        SPVariables.setString(CameraPreview, camerapre, SettingActivity.this);

        startActivity(new Intent(SettingActivity.this, SettingActivity.class).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION));
        finish();
        overridePendingTransition(0, 0);

    }

    public void Rating(View view) {

        //Rating bar code

        /*Uri uri = Uri.parse("market://details?id=" + getPackageName());
        Intent myAppLinkToMarket = new Intent(Intent.ACTION_VIEW, uri);
        try {
            startActivity(myAppLinkToMarket);
        } catch (Exception e) {
            Toast.makeText(this, " unable to find market app", Toast.LENGTH_LONG).show();
        }*/


        //Please check this out upwards code
        Toast.makeText(this, "Coming Soon...", Toast.LENGTH_SHORT).show();


    }

    public void Sharethelink(View view) {


        //Please check this out upwards code
        Toast.makeText(this, "Coming Soon...", Toast.LENGTH_SHORT).show();


    }
}

File: app/src/main/java/com/manddprojectconsultant/screencam/activity/SettingActivity.kt
package com.manddprojectconsultant.screencam.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.manddprojectconsultant.screencam.BuildConfig
import com.manddprojectconsultant.screencam.R
import com.manddprojectconsultant.screencam.databinding.ActivitySettingBinding
import com.manddprojectconsultant.screencam.fragment.CameraFacingFragment
import com.manddprojectconsultant.screencam.fragment.CameraFacingFragment.ItemClickListenercamerafacing
import com.manddprojectconsultant.screencam.fragment.CameraPreviewFragment
import com.manddprojectconsultant.screencam.fragment.CameraPreviewFragment.ItemClickListenercamerapreview
import com.manddprojectconsultant.screencam.fragment.FPSFragment
import com.manddprojectconsultant.screencam.fragment.FPSFragment.ItemClickListenerfps
import com.manddprojectconsultant.screencam.fragment.OrientationFragment
import com.manddprojectconsultant.screencam.fragment.OrientationFragment.ItemClickListenerorientation
import com.manddprojectconsultant.screencam.fragment.QualitySheetFragment
import com.manddprojectconsultant.screencam.fragment.QualitySheetFragment.ItemClickListenervideo
import com.manddprojectconsultant.screencam.utils.PreferenceManager
import com.manddprojectconsultant.screencam.utils.SPVariables
import com.manddprojectconsultant.screencam.widget.ResolutionSheetDialog
import com.manddprojectconsultant.screencam.widget.ResolutionSheetDialog.ItemClickListener
import androidx.core.net.toUri

class SettingActivity : AppCompatActivity(), ItemClickListenercamerapreview,
    ItemClickListenercamerafacing, ItemClickListenerorientation, ItemClickListenerfps,
    ItemClickListenervideo, ItemClickListener {
        lateinit var binding: ActivitySettingBinding
//    private var binding.llResolution: RelativeLayout? = null
//    private var binding.llQuality: RelativeLayout? = null
//    private var binding.llFPS: RelativeLayout? = null
//    private var binding.llOrientation: RelativeLayout? = null
//    private var binding.llRecordAudio: RelativeLayout? = null
//    private var binding.llFileLocation: RelativeLayout? = null
//    private var binding.llCountDown: RelativeLayout? = null
//    private var binding.llCamera: RelativeLayout? = null
//    private var binding.llCameraPreview: RelativeLayout? = null
//    private var binding.llShowBubble: RelativeLayout? = null
//    private val llRating: RelativeLayout? = null
//    private val llShare: RelativeLayout? = null
//    private var binding.tvResolution: TextView? = null
//    private var binding.tvQuality: TextView? = null
//    private var binding.tvFPS: TextView? = null
//    private var binding.tvOrientation: TextView? = null
//    private var binding.tvCamera: TextView? = null
//    private var binding.tvCameraPreview: TextView? = null
//    private var binding.swRecordAudio: Switch? = null
//    private var binding.swCountDown: Switch? = null
//    private val swAppIntro: Switch? = null
//    private var binding.swShowBubble: Switch? = null
//    private var binding.llAppIntro: ImageView? = null
//    private var binding.ivbackbutton: ImageView? = null

    //final String[] list = getActivity().getResources().getStringArray(R.array.Choice_items);
    private var resolutionList: Array<String> = arrayOf()
    private var qualityList: Array<String> = arrayOf()
    private var fpsList: Array<String> = arrayOf()
    private var orientationList: Array<String> = arrayOf()
    private var cameraList: Array<String> = arrayOf()
    private var cameraListPreview: Array<String> = arrayOf()

    // public static final String RecordAudio = "record audio";
    private var qualitySheetFragment: QualitySheetFragment = QualitySheetFragment()
    private var resolutionSheetDialog: ResolutionSheetDialog = ResolutionSheetDialog()
    private var fpsFragment: FPSFragment = FPSFragment()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingBinding.inflate(layoutInflater)
        setContentView(binding.root)
        findAllView()


        //Resolution Preference
        val resolution = SPVariables.getString(Resolution, this@SettingActivity)
        binding.tvResolution.text = resolution

        //Video Quality Preference
        val videoquality = SPVariables.getString(VideoQuality, this@SettingActivity)
        binding.tvQuality.text = videoquality

        //FPS preference
        val fpspre = SPVariables.getString(FPS, this@SettingActivity)
        binding.tvFPS.text = fpspre

        //Orientation preference
        val orientationpre = SPVariables.getString(Orientation, this@SettingActivity)
        binding.tvOrientation.text = orientationpre

        //CameraFacing preference
        val camerafacingpre = SPVariables.getString(cameraFacing, this@SettingActivity)
        binding.tvCamera.text = camerafacingpre


        //CameraPreview Preference
        val camerapreviewpre = SPVariables.getString(cameraPreview, this@SettingActivity)
        binding.tvCameraPreview.text = camerapreviewpre





        title = "Settings"
        allList
        loadSettingDataFromSP()
        allOnClicks()
    }

    private fun loadSettingDataFromSP() {
        //Resolution
        val aa = SPVariables.getString(Resolution, this@SettingActivity)
        if (SPVariables.getString("Resolution", this@SettingActivity).isEmpty()) {
            binding.tvResolution.text = aa
        } else {
            //binding.tvResolution.setText(SPVariables.getString("resolution", SettingActivity.this));
            binding.tvResolution.text =
                SPVariables.getString("Resolution", this@SettingActivity)
        }

        //Video Quality
        val videoquality = SPVariables.getString(VideoQuality, this@SettingActivity)
        if (SPVariables.getString("Quality", this@SettingActivity).isEmpty()) {
            binding.tvQuality.text = videoquality
        } else {
            binding.tvQuality.text = SPVariables.getString("Quality", this@SettingActivity)
        }

        //FPS
        val fpspre = SPVariables.getString(FPS, this@SettingActivity)
        if (SPVariables.getString("FPS", this@SettingActivity).isEmpty()) {
            binding.tvFPS.text = fpspre
        } else {
            binding.tvFPS.text = SPVariables.getString("FPS", this@SettingActivity)
        }

        //Orientation
        val orientationpre = SPVariables.getString(Orientation, this@SettingActivity)
        if (SPVariables.getString("Orientation", this@SettingActivity).isEmpty()) {
            binding.tvOrientation.text = orientationpre
        } else {
            binding.tvOrientation.text =
                SPVariables.getString("Orientation", this@SettingActivity)
        }

        //Camera Facing
        val camerafacingpre = SPVariables.getString(cameraFacing, this@SettingActivity)
        if (SPVariables.getString("CameraFacing", this@SettingActivity).isEmpty()) {
            binding.tvCamera.text = camerafacingpre
        } else {
            binding.tvCamera.text =
                SPVariables.getString("CameraFacing", this@SettingActivity)
        }

        //Camera Preview
        val camerapreviewpre = SPVariables.getString(cameraPreview, this@SettingActivity)
        if (SPVariables.getString("CameraPreview", this@SettingActivity).isEmpty()) {
            binding.tvCameraPreview.text = camerapreviewpre
        } else {
            binding.tvCameraPreview.text =
                SPVariables.getString("CameraPreview", this@SettingActivity)
        }

        if (SPVariables.getString("RecordAudio", this@SettingActivity) == "TRUE") {
            binding.swRecordAudio.isChecked = true
        } else {
            binding.swRecordAudio.isChecked = false
        }

        if (SPVariables.getString("CountDown", this@SettingActivity) == "TRUE") {
            binding.swCountDown.isChecked = true
        } else {
            binding.swCountDown.isChecked = false
        }
        //        if (SPVariables.getString("AppIntro", SettingActivity.this).equals("TRUE")) {
//            swAppIntro.setChecked(true);
//        } else {
//            swAppIntro.setChecked(false);
//        }
        if (SPVariables.getString("ShowBubble", this@SettingActivity) == "TRUE") {
            binding.swShowBubble.isChecked = true
        } else {
            binding.swShowBubble.isChecked = false
        }
    }

    private fun allOnClicks() {
        binding.llResolution.setOnClickListener { /*DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(ResolutionList, "Resolution", binding.tvResolution.getText().toString());
                    SingleChoiceDialog.setCancelable(true);
                    SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
            resolutionSheetDialog = ResolutionSheetDialog()
            resolutionSheetDialog.show(supportFragmentManager, "Resolution")
        }
        binding.llQuality.setOnClickListener { /*     DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(QualityList, "Quality", binding.tvQuality.getText().toString());
                    SingleChoiceDialog.setCancelable(true);
                    SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
            qualitySheetFragment = QualitySheetFragment()
            qualitySheetFragment.show(supportFragmentManager, "Quality")
        }
        binding.llFPS.setOnClickListener { /*DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(FPSList, "FPS", binding.tvFPS.getText().toString());
                    SingleChoiceDialog.setCancelable(true);
                    SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
            fpsFragment = FPSFragment()
            fpsFragment.show(supportFragmentManager, "FPS")
        }
        binding.llOrientation.setOnClickListener { /* DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(OrientationList, "Orientation", binding.tvOrientation.getText().toString());
                    SingleChoiceDialog.setCancelable(true);
                    SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
            val orientationFragment = OrientationFragment()
            orientationFragment.show(supportFragmentManager, "Orientation")
        }
        binding.llCamera.setOnClickListener { /*        DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(CameraList, "Camera", binding.tvCamera.getText().toString());
                    SingleChoiceDialog.setCancelable(true);
                    SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
            val cameraFacingFragment = CameraFacingFragment()
            cameraFacingFragment.show(supportFragmentManager, "CameraFacing")
        }
        binding.llCameraPreview.setOnClickListener { /*        DialogFragment SingleChoiceDialog = new SingleChoiceDialogFragment(CameraListPreview, "CameraPreview", binding.tvCameraPreview.getText().toString());
                    SingleChoiceDialog.setCancelable(true);
                    SingleChoiceDialog.show(getSupportFragmentManager(), "Single Choice Dialog");*/
            val cameraPreviewFragment = CameraPreviewFragment()
            cameraPreviewFragment.show(supportFragmentManager, "Camera Preview")
        }






        binding.swRecordAudio.setOnCheckedChangeListener { buttonView, isChecked ->
            if (isChecked) {
                SPVariables.setString("RecordAudio", "TRUE", this@SettingActivity)
            } else {
                SPVariables.setString("RecordAudio", "FALSE", this@SettingActivity)
            }
        }
        binding.swCountDown.setOnCheckedChangeListener { buttonView, isChecked ->
            if (isChecked) {
                SPVariables.setString("CountDown", "TRUE", this@SettingActivity)
            } else {
                SPVariables.setString("CountDown", "FALSE", this@SettingActivity)
            }
        }
        binding.llAppIntro.setOnClickListener {
            val preferenceManager =
                PreferenceManager(this@SettingActivity)
            preferenceManager.isFirstTimeLaunch = true
            val intent = Intent(this@SettingActivity, WelcomeActivity::class.java)
            startActivityForResult(intent, REQUEST_CODE_INTRO)
            //finish();
        }
        //        swAppIntro.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                if (isChecked) {
//                    SPVariables.setString("AppIntro", "TRUE", SettingActivity.this);
//                } else {
//                    SPVariables.setString("AppIntro", "FALSE", SettingActivity.this);
//                }
//            }
//        });
        binding.swShowBubble.setOnCheckedChangeListener { buttonView, isChecked ->
            if (isChecked) {
                SPVariables.setString("ShowBubble", "TRUE", this@SettingActivity)
            } else {
                SPVariables.setString("ShowBubble", "FALSE", this@SettingActivity)
            }
        }
        binding.ivbackbutton.setOnClickListener { onBackPressed() }
    }

    override fun onBackPressed() {
        val intent = Intent(this@SettingActivity, DashboardActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        startActivity(intent)
        overridePendingTransition(0, 0)
        finish()
        super.onBackPressed()
    }

    private val allList: Unit
        get() {
            resolutionList = resources.getStringArray(R.array.Resolution)
            qualityList = resources.getStringArray(R.array.Quality)
            fpsList = resources.getStringArray(R.array.FPS)
            orientationList = resources.getStringArray(R.array.Orientation)
            cameraList = resources.getStringArray(R.array.Camera)
            cameraListPreview = resources.getStringArray(R.array.CameraPreview)
        }

    private fun findAllView() {
//        binding.ivbackbutton = findViewById(R.id.binding.ivbackbutton)
//        binding.llResolution = findViewById(R.id.binding.llResolution)
//        binding.llQuality = findViewById(R.id.binding.llQuality)
//        binding.llFPS = findViewById(R.id.binding.llFPS)
//        binding.llOrientation = findViewById(R.id.binding.llOrientation)
//        binding.llRecordAudio = findViewById(R.id.binding.llRecordAudio)
//        binding.llFileLocation = findViewById(R.id.binding.llFileLocation)
//        binding.llCountDown = findViewById(R.id.binding.llCountDown)
//        binding.llCamera = findViewById(R.id.binding.llCamera)
//        binding.llCameraPreview = findViewById(R.id.binding.llCameraPreview)
//        binding.llAppIntro = findViewById(R.id.binding.llAppIntro)
//        binding.llShowBubble = findViewById(R.id.binding.llShowBubble)
//        binding.tvResolution = findViewById(R.id.binding.tvResolution)
//        binding.tvQuality = findViewById(R.id.binding.tvQuality)
//        binding.tvFPS = findViewById(R.id.binding.tvFPS)
//        binding.tvOrientation = findViewById(R.id.binding.tvOrientation)
//        binding.tvCamera = findViewById(R.id.binding.tvCamera)
//        binding.tvCameraPreview = findViewById(R.id.binding.tvCameraPreview)
//        binding.swRecordAudio = findViewById(R.id.binding.swRecordAudio)
//        binding.swCountDown = findViewById(R.id.binding.swCountDown)
//        //swAppIntro = findViewById(R.id.swAppIntro);
//        binding.swShowBubble = findViewById(R.id.binding.swShowBubble)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_CODE_INTRO) {
            if (resultCode == RESULT_OK) {
                android.preference.PreferenceManager.getDefaultSharedPreferences(this).edit()
                    .putBoolean(PREF_KEY_FIRST_START, false)
                    .apply()
            } else {
                android.preference.PreferenceManager.getDefaultSharedPreferences(this).edit()
                    .putBoolean(PREF_KEY_FIRST_START, true)
                    .apply()
                //User cancelled the intro so we'll finish this activity too.
                //finish();
            }
        }
    }

    override fun onItemClick(item: String) {
        val resolution = item
        SPVariables.setString(Resolution, resolution, this@SettingActivity)
        startActivity(
            Intent(
                this@SettingActivity,
                SettingActivity::class.java
            ).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        )
        finish()
        overridePendingTransition(0, 0)
        //binding.tvResolution.setText(resolution);
    }

    override fun onItemClickvideo(videoquality: String) {
        val vidquality = videoquality
        SPVariables.setString(VideoQuality, vidquality, this@SettingActivity)
        startActivity(
            Intent(
                this@SettingActivity,
                SettingActivity::class.java
            ).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        )
        finish()
        overridePendingTransition(0, 0)
    }

    override fun onItemClickfps(fps: String) {
        val fp = fps
        SPVariables.setString(FPS, fp, this@SettingActivity)
        startActivity(
            Intent(
                this@SettingActivity,
                SettingActivity::class.java
            ).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        )
        finish()
        overridePendingTransition(0, 0)
    }

    override fun onItemClickorientation(orientation: String) {
        val orien = orientation
        SPVariables.setString(Orientation, orien, this@SettingActivity)
        startActivity(
            Intent(
                this@SettingActivity,
                SettingActivity::class.java
            ).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        )
        finish()
        overridePendingTransition(0, 0)
    }

    override fun onItemClickcamerafacing(camerafacing: String) {
        val camera = camerafacing
        SPVariables.setString(cameraFacing, camera, this@SettingActivity)
        startActivity(
            Intent(
                this@SettingActivity,
                SettingActivity::class.java
            ).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        )
        finish()
        overridePendingTransition(0, 0)
    }

    override fun onItemClickcamerapreview(camerapreview: String) {
        val camerapre = camerapreview

        SPVariables.setString(cameraPreview, camerapre, this@SettingActivity)

        startActivity(
            Intent(
                this@SettingActivity,
                SettingActivity::class.java
            ).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        )
        finish()
        overridePendingTransition(0, 0)
    }

    fun Rating() {
        //Rating bar code

        val uri = "market://details?id=$packageName".toUri()
        val myAppLinkToMarket = Intent(Intent.ACTION_VIEW, uri)
        try {
            startActivity(myAppLinkToMarket)
        } catch (e: Exception) {
            Toast.makeText(this, " unable to find market app", Toast.LENGTH_LONG).show()
        }


        /*//Please check this out upwards code
        Toast.makeText(this, "Coming Soon...", Toast.LENGTH_SHORT).show();
*/
    }

    fun shareTheLink() {
        try {
            val shareIntent = Intent(Intent.ACTION_SEND)
            shareIntent.setType("text/plain")
            shareIntent.putExtra(Intent.EXTRA_SUBJECT, "Screen Cam")
            var shareMessage = "\nPlease recommend this app to your circle.\n"
            shareMessage =
                shareMessage + "https://play.google.com/store/apps/details?id=" + BuildConfig.APPLICATION_ID + "\n\n"
            shareIntent.putExtra(Intent.EXTRA_TEXT, shareMessage)
            startActivity(Intent.createChooser(shareIntent, "choose one"))
        } catch (e: Exception) {
            //e.toString();
        }
    }

    companion object {
        const val REQUEST_CODE_INTRO: Int = 1
        const val PREF_KEY_FIRST_START: String = "PREF_KEY_FIRST_START"
        const val Resolution: String = "Resolution"
        const val VideoQuality: String = "Quality"
        const val FPS: String = "FPS"
        const val Orientation: String = "Orientation"
        const val cameraFacing: String = "CameraFacing"
        const val cameraFrame: String = "CameraFrame"
        const val cameraPreview: String = "CameraPreview"
    }
}
